lib/matrix_convert_to_zero.rb

```diff
@@ -3,8 +3,57 @@
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
 
-# Time complexity: ?
-# Space complexity: ?
+#Approach 1:
+# Time complexity: O(n^3) 3 nested loops
+# Space complexity: O(n^2) n is the number of rows and columns in the matrix being cloned
+
+def matrix_convert_to_zero_n_3(matrix)
+  clone = matrix.map(&:dup)
+  rows = matrix.length
+  columns = matrix[0].length
+  rows.times do |i|
+    columns.times do |j|
+      if matrix[i][j] == 0
+        columns.times do |jj|
+          clone[i][jj] = 0
+        end
+        rows.times do |ii|
+          clone[ii][j] = 0
+        end
+      end
+    end
+  end
+  rows.times do |i|
+    columns.times do |j|
+      matrix[i][j] = clone[i][j]
+    end
+  end
+end
+
+#Approach 2:
+# Time complexity: O(n^2) 2 nested loops
```

Comment:
Same comment as above (`O(nm)` instead of `O(n^2)`), but well done! This second implementation is a significant improvement over the `O(n^3)` implementation above.